internal/buffer/unbounded.go

```diff
@@ -72,14 +79,25 @@ func (b *Unbounded) Load() {
 		default:
 		}
 	}
-	b.mu.Unlock()
 }
 
 // Get returns a read channel on which values added to the buffer, via Put(),
 // are sent on.
 //
 // Upon reading a value from this channel, users are expected to call Load() to
 // send the next buffered value onto the channel if there is any.
+// If the unbounded buffer is closed, a read channel returned by this method is closed.
```

Comment:
Nit: can you wrap this additional comment to 80 characters (and probably put it on a separate sentence? Also switch "a read channel" to "the read channel".